src/Protocol/OStatus.php

```diff
@@ -686,6 +673,25 @@ private static function processPost($xpath, $entry, &$item, $importer)
 			$item["parent-uri"] = $item["uri"];
 		}
 
+		$categories = $xpath->query('atom:category', $entry);
+		if ($categories) {
+			foreach ($categories as $category) {
+				foreach ($category->attributes as $attributes) {
+					if ($attributes->name == "term") {
+						$term = $attributes->textContent;
+						// don't add nsfw with content warning
+						// Background: "nsfw" is set automatically by Mastodon and superfluous
```

Comment:
How is it superfluous? We still aren’t supporting Mastodon “Sensitive Image” flag that I know of?

Reply:
"Content Warning" is supported. And currently you had to first click on the "nsfw" enlargement and then on the "content warning" enlargement to see the content.

Reply:
AFAIK, Content Warning is different from sensitive content. The first is like a spoiler tag for text, while the second is a spoiler tag specifically for images, see #4414.

Reply:
This change is not about "sensitive content". It is just about the "content warning" that always adds this "nsfw" category.

So now you see this post:
![bildschirmfoto 2018-03-14 um 01 35 01](https://user-images.githubusercontent.com/844208/37377120-704fba42-2728-11e8-96bc-c9fef0cc577e.png)

You click on the link:
![bildschirmfoto 2018-03-14 um 01 35 17](https://user-images.githubusercontent.com/844208/37377129-7c57eb66-2728-11e8-8df1-90b26b7555a1.png)

And after you clicked on the next link, you finally see the content:
![bildschirmfoto 2018-03-14 um 01 35 27](https://user-images.githubusercontent.com/844208/37377144-8a19fa8c-2728-11e8-8c54-895233d8c651.png)

Because of the change it now looks this way:
![bildschirmfoto 2018-03-14 um 01 35 54](https://user-images.githubusercontent.com/844208/37377154-97ebea8a-2728-11e8-861b-1bce8265c6ad.png)
![bildschirmfoto 2018-03-14 um 01 40 39](https://user-images.githubusercontent.com/844208/37377192-bf4a96d0-2728-11e8-9716-d904d3156c53.png)

And of course the "nsfw" is only removed when there was a content warning.

Reply:
The NSFW hiding is provided by an addon so I'm against removing a category in a Friendica core protocol because it makes a problem with an addon. If anything, the addon should be improved.

Reply:
The addon doesn't have all the information at that time. I guess the others should decide here. @tobiasd, @fabrixxm, @rabuzarus - what do you think?

Reply:
I agree with @MrPetovan , if thy want to have dirty talk about everything (which they do by sending the NSFW tag along) they should do and we should respect this.

Reply:
With the config setting the Friendica admins could decide individually if they consider this a bug or a feature.